Products/CMFPlone/browser/search.py

```diff
@@ -221,6 +221,11 @@ def navroot_url(self):
             self._navroot_url = state.navigation_root_url()
         return self._navroot_url
 
+    def current_section_title(self):
+        if 'path' in self.request:
+            return self.context.unrestrictedTraverse(self.request['path']).Title()
```

Comment:
Uhm... I do not like this unrestrictedTraverse, also what happens if self.request['path'] is not a valid path?
Maybe 
```
section = self.context.restrictedTraverse(self.request['path'], None)
if section:
    return section.Title()
```
is better

Reply:
Fixed